spec/models/vim_performance_state_spec.rb

```diff
@@ -20,6 +55,45 @@
     end
   end
 
+  describe "#containers" do
+    it "doesnt return records that are archived before the range" do
+      container_image = FactoryBot.create(:container_image)
+      # not deleted is captured
+      containers = FactoryBot.create_list(:container, 1, :container_image => container_image)
+      # deleted before is not captured
+      other_containers = FactoryBot.create_list(:container, 1, :container_image => container_image, :deleted_on => (capture_time - 2.hours))
+      # deleted later is captured
+      containers << FactoryBot.create(:container, :container_image => container_image, :deleted_on => (capture_time + 2.hours))
+
+      state_data = {:assoc_ids => {:containers => {:on => (containers + other_containers).map(&:id)}}}
+      actual = described_class.new(:timestamp => capture_time, :state_data => state_data)
+      expect(actual.containers.sort).to eq(containers.sort)
+    end
+
+    it "handles pruned ids" do
+      container_image = FactoryBot.create(:container_image)
+      # not deleted is captured
+      containers = FactoryBot.create_list(:container, 1, :container_image => container_image)
+
+      state_data = {:assoc_ids => {:containers => {:on => (containers.map(&:id) + [22, 23, 24, 25, 26, 27])}}}
```

Comment:
What are the random numbers for?  If it's to get "other" ids, prefer building the id list from existing records, because a) this doesn't handle multi-region properly where the ids aren't in the 0 range and b) there's a chance the existing records could fall into the hardcoded range and fail the test.

Reply:
put in a comment.

these represent ids that are no longer valid.
If container.id is within that range, then one of those ids will work, but the others will be thrown away. so there will be no false positives in here.

I could do `containers.last.id` and build from there? - but that seemed like a bunch of code that will create invalid ids just as easily as this does.

Reply:
`container.last.id` sounds reasonable - we do this in other specs as well.  We've hit problems in other places where we fake ids for records that either don't exist or shouldn't exist and hardcoded ids have always eventually caused problems.